tests/test_cases/test_cocotb/test_async_coroutines.py

```diff
@@ -135,3 +135,45 @@ async def example():
 
     yield example()
     assert ran
+
+
+@cocotb.test()
+async def test_fork_coroutine_function_exception(dut):
+    async def coro():
+        pass
+
+    try:
+        cocotb.fork(coro)
+        assert False
+    except TypeError as e:
+        cocotb.log.info("error: {}".format(e))
+        assert str(e) == "Coroutine function {} should be called " \
+            "prior to being scheduled.".format(coro)
+
+
+@cocotb.test()
+async def test_factory_coroutine_function_exception(dut):
+    async def coro(dut):
+        pass
+
+    try:
+        cocotb.regression.TestFactory(coro)
+        assert False
+    except TypeError as e:
+        cocotb.log.info("error: {}".format(e))
+        assert str(e) == "Coroutine function {} should be called " \
+            "prior to being scheduled.".format(coro)
+
+
+@cocotb.test()
+async def test_task_coroutine_function_exception(dut):
+    async def coro(dut):
+        pass
+
+    try:
+        cocotb.decorators.RunningTask(coro)
+        assert False
+    except TypeError as e:
+        cocotb.log.info("error: {}".format(e))
```

Comment:
The log messages you have here are not so useful for unit tests. Perhaps knock them down to debug.